langgraph/checkpoint/base.py

```diff
@@ -93,8 +98,27 @@ class CheckpointTuple(NamedTuple):
 )
 
 
-class BaseCheckpointSaver(Serializable, ABC):
-    at: CheckpointAt = CheckpointAt.END_OF_RUN
+class SerializerProtocol(Protocol):
+    def dumps(self, obj: Any) -> bytes:
+        ...
+
+    def loads(self, data: bytes) -> Any:
+        ...
+
+
+class BaseCheckpointSaver(ABC):
+    at: CheckpointAt = CheckpointAt.END_OF_STEP
+
+    serde: SerializerProtocol
```

Comment:
so this is for specifying an inheritable default?

Reply:
you mean the previous line?